From f95897cef614fff710c31d9e478eacc85d6312d5 Mon Sep 17 00:00:00 2001
From: Peter Oh <peter.oh@bowerswilkins.com>
Date: Tue, 29 May 2018 14:39:18 -0700
Subject: [PATCH 14/18] mesh: do not allow scan result to swap pri/sec

Swapping between primary and secondary channel will break
mesh from joining, hence don't allow it.

Signed-off-by: Peter Oh <peter.oh@bowerswilkins.com>
---
 wpa_supplicant/wpa_supplicant.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- a/wpa_supplicant/wpa_supplicant.c
+++ b/wpa_supplicant/wpa_supplicant.c
@@ -2185,7 +2185,7 @@ void ibss_mesh_setup_freq(struct wpa_sup
 	}
 	freq->sec_channel_offset = ht40;
 
-	if (obss_scan) {
+	if (ssid->mode != WPAS_MODE_MESH && obss_scan) {
 		struct wpa_scan_results *scan_res;
 
 		scan_res = wpa_supplicant_get_scan_results(wpa_s, NULL, 0);
